### DISCUSSION TITLE
Inventor crashing using GetFileByFullFilePath

### DISCUSSION THREAD

**Discussion by GiliamC** [koechlm/iLogic-Vault#23] [Q&A]
Hello,

I am using the QuickstartiLogicLibrary, Version=2023.0.0.0, and sometimes Inventor crashes when using the function GetFileByFullFilePath(oPath, False). I am using Inventor Build 418 Release 2023.4.1.

I am using the function in an assembly to retrieve all the .idw files from the parts/sub-assemblies in the assembly.
It did not happen in inventor 2019

Hopefully, you can help me. In any case, thank you!

See code below:

For Each oRefDoc In oRefDocs
            If oRefDoc.DocumentType = DocumentTypeEnum.kPartDocumentObject Then
                oFilename = Replace(oRefDoc.FullFileName, ".ipt", ".idw")
                oExtensie = ".ipt"
            ElseIf oRefDoc.DocumentType = DocumentTypeEnum.kAssemblyDocumentObject Then
                oFilename = Replace(oRefDoc.FullFileName, ".iam", ".idw")
                oExtensie = ".iam"
            End If

            Try
                SharedClass.Get_File_From_Vault(oFilename) 
            Catch
                MsgBox(oFilename & " Foutmelding!")
            End Try

            
        Next

'Shared Class
    Public Shared iLogicVault As New QuickstartiLogicLibrary.QuickstartiLogicLib

   Public Shared Sub Get_File_From_Vault(oPad As String)
        Dim mVaultConnection As VDF.Vault.Currency.Connections.Connection
        mVaultConnection = VB.ConnectionManager.Instance.Connection

        If mVaultConnection Is Nothing Then
            MessageBox.Show("Niet ingelogd in de Vault! - Log eerst in.")
            Exit Sub
        End If

        Dim VaultPath As String = oPad
        VaultPath = VaultPath.Replace("C:\Workspace\", "$/")
        VaultPath = VaultPath.Replace("\", "/")

        'Get File to C-schijf
        Try
            Dim mVaultFile As String = iLogicVault.GetFileByFullFilePath(VaultPath, False)
        Catch
            MsgBox("Niet Gelukt                " & VaultPath)
        End Try

    End Sub

**Comment by GiliamC** [comment]
Inventor Error Report


Date/Time: 2023-12-08 08:51:14 +01:00
Application: Inventor.exe
Error: C++ EH exception - code e06d7363 (first/second chance not available)
Crashed Module Name: KERNELBASE.dll
Exception Address: 0x00007ff8e902cf19
Exception Code: e06d7363
Exception Flags: 1
Exception Parameters: 19930520, 596077a330, 7ff892abbb80, 7ff892900000

**Reply by koechlm** [reply]
@GiliamC : as you are handling the file retrieval using try-catch I think the crash is not related to the call itself rather than the state of one of the files in memory. The file retrieval via iLogic is revision based and enforcing override, so any changes of your models in memory that are not checked-in before you get the drawings may cause conflicts. Please help to identify a reproducible state/action scenario. The first step might be to change the drawing retrieval option to check-out = true.

**Reply by GiliamC** [reply]
Hello, thank you for the quick response!

The activation of the checkout function has unfortunately not made any difference. 
Furthermore, does the function currently change the vault status of parts? See the images below. In this case, it did not crash. 
It seems like the function is attempting to retrieve an old version from the vault.

Additionally, I think it's related to library components. We are working with a separate library containing standard parts.
We design custom parts for projects(LANDD folder) along with standard components from the library 
![image](https://github.com/koechlm/iLogic-Vault/assets/153267133/e5007c3a-fa72-4013-b978-f463c0ed93ca)


Btw, I am using Visual studio 2019 for the code.

When I open the QuickstartiLogicLibrary, it crashes on the following line:
//download
VDF.Vault.Results.AcquireFilesResults results = conn.FileManager.AcquireFiles(settings);

Before
![image](https://github.com/koechlm/iLogic-Vault/assets/153267133/802cd45b-d842-4947-855f-b9e6f814a2fb)
After
![image](https://github.com/koechlm/iLogic-Vault/assets/153267133/2b636219-26ef-4192-b936-c56da9e09d30)
